### PR TITLE
Entity AI mappings

### PR DESCRIPTION
**Pull request by IotaBread**
Mostly entity AI mappings, plus some constants in `HungerConstants`

### CODE REVIEW

**Comment by triphora** [comment]
`EXHAUSTION` is misspelled in several places here as `EXAHUSTION`

File: mappings/net/minecraft/entity/ai/control/LookControl.mapping [hunk 1]
```diff
@@ -2,18 +2,24 @@ CLASS net/minecraft/unmapped/C_lgufpcrf net/minecraft/entity/ai/control/LookCont
...
+		ARG 1 pitch
 	METHOD m_dpevsitq lookAt (Lnet/minecraft/unmapped/C_vgpupfxx;)V
 		ARG 1 direction
 	METHOD m_duxqnltr lookAt (DDD)V
 		ARG 1 x
 		ARG 3 y
 		ARG 5 z
+	METHOD m_epvguicl (Ljava/lang/Float;)V
+		ARG 1 yaw
+	METHOD m_gjtczvwx clampHeadYAw ()V
```
Comment (triphora) [inline mappings/net/minecraft/entity/ai/control/LookControl.mapping]:
```suggestion
	METHOD m_gjtczvwx clampHeadYaw ()V
```

File: mappings/net/minecraft/entity/ai/control/MoveControl.mapping [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
 	FIELD f_amdgmehs targetX D
+	FIELD f_bllciqqk MIN_SPEED_SQUARED F
+	FIELD f_btsujdzc MAX_TURN I
 	FIELD f_cpgduhem targetZ D
 	FIELD f_hoicqoxx sidewaysMovement F
 	FIELD f_krcwsjhv forwardMovement F
 	FIELD f_mcknyygq targetY D
 	FIELD f_qvjczzzo state Lnet/minecraft/unmapped/C_azhvkebm$C_rcgaukie;
 	FIELD f_vikyuanr entity Lnet/minecraft/unmapped/C_dxkfswlz;
+	FIELD f_xllxyxuw MIN_SPPED F
```
Comment (triphora) [inline mappings/net/minecraft/entity/ai/control/MoveControl.mapping]:
```suggestion
	FIELD f_xllxyxuw MIN_SPEED F
```

**Review by triphora — APPROVED** [review]
Just typos

File: mappings/net/minecraft/entity/ai/pathing/BirdNavigation.mapping [hunk 3]
```diff
@@ -1,5 +1,6 @@
 CLASS net/minecraft/unmapped/C_rftzegur net/minecraft/entity/ai/pathing/BirdNavigation
```
Comment (OroArmor) [inline mappings/net/minecraft/entity/ai/pathing/BirdNavigation.mapping:1]: Hmm, dont know what i think about this class name.

**Review by OroArmor — APPROVED** [review]

**Review by LambdAurora — APPROVED** [review]

**Review by NoComment1105 — APPROVED** [review]

File: mappings/net/minecraft/entity/ai/brain/task/TemptTask.mapping [hunk 4]
```diff
@@ -1,4 +1,5 @@
 CLASS net/minecraft/unmapped/C_jazltyri net/minecraft/entity/ai/brain/task/TemptTask
+	FIELD f_jicdluie CLOSE_ENOUGH_DISTANCE D
```
Comment (Bubblie01) [inline mappings/net/minecraft/entity/ai/brain/task/TemptTask.mapping:2]: MINIMUM_DISTANCE

File: mappings/net/minecraft/entity/ai/brain/task/WalkTask.mapping [hunk 5]
```diff
@@ -1,7 +1,16 @@
 CLASS net/minecraft/unmapped/C_uuyjertq net/minecraft/entity/ai/brain/task/WalkTask
 	FIELD f_bkeyoylk speed F
+	FIELD f_fqkxixwg HORIZONTAL_RANGE I
 	FIELD f_iynajoye MAX_RUN_TIME I
 	FIELD f_trxuqcna VERTICAL_RANGE I
 	FIELD f_wcvudbwu MIN_RUN_TIME I
 	METHOD <init> (F)V
 		ARG 1 speed
+	METHOD m_dbqqkxir findClosestWater (Lnet/minecraft/unmapped/C_peaveboq;Lnet/minecraft/unmapped/C_astfners;)Ljava/util/Optional;
```
Comment (Bubblie01) [inline mappings/net/minecraft/entity/ai/brain/task/WalkTask.mapping:9]: findWater

File: mappings/net/minecraft/entity/ai/control/LookControl.mapping [hunk 6]
```diff
@@ -2,18 +2,24 @@ CLASS net/minecraft/unmapped/C_lgufpcrf net/minecraft/entity/ai/control/LookCont
 	FIELD f_dcgmdxjd pitchSpeed F
 	FIELD f_hlcqilwt lookY D
 	FIELD f_jptovgyv lookX D
+	FIELD f_ojwuhwmo lootAtCooldown I
```
Comment (Bubblie01) [inline mappings/net/minecraft/entity/ai/control/LookControl.mapping:5]: SPELLING ERROR: lookAtCooldown

Comment (Bubblie01) [reply mappings/net/minecraft/entity/ai/control/LookControl.mapping:5]: BETTER NAME: lookCooldown

File: mappings/net/minecraft/entity/ai/goal/EscapeDangerGoal.mapping [hunk 7]
```diff
@@ -5,13 +5,16 @@ CLASS net/minecraft/unmapped/C_gilizaew net/minecraft/entity/ai/goal/EscapeDange
 	FIELD f_hpmxvyiq targetZ D
 	FIELD f_najlgwjk targetY D
 	FIELD f_svbrvkbw speed D
+	FIELD f_ypymgqlx WATER_CHECK_DISTANCE_VERTICAL I
```
Comment (Bubblie01) [inline mappings/net/minecraft/entity/ai/goal/EscapeDangerGoal.mapping:8]: WATER_CHECK_VERTICAL_DISTANCE

File: mappings/net/minecraft/entity/ai/goal/GoalSelector.mapping [hunk 8]
```diff
@@ -30,11 +36,18 @@ CLASS net/minecraft/unmapped/C_qospjrjn net/minecraft/entity/ai/goal/GoalSelecto
...
 		ARG 2 enabled
 	METHOD m_ikdikwii getRunningGoals ()Ljava/util/stream/Stream;
+	METHOD m_jgxhmcyj tickGoals (Z)V
+		ARG 1 tickAll
 	METHOD m_jgygovod tick ()V
 	METHOD m_lfaldjmu clear ()V
 	METHOD m_mhoeonji getGoals ()Ljava/util/Set;
 	METHOD m_puiwaruk setTimeInterval (I)V
 		ARG 1 timeInterval
+	METHOD m_ugwtceby goalUsesAnyControl (Lnet/minecraft/unmapped/C_wgswcjhs;Ljava/util/EnumSet;)Z
```
Comment (Bubblie01) [inline mappings/net/minecraft/entity/ai/goal/GoalSelector.mapping:46]: isGoalUsingControl or goalUsesControl

**Review by Bubblie01 — COMMENTED** [review]
Please take a look at the comments I provided, there was also a spelling error

**Pull request merged**